src/town_cmd.cpp

```diff
@@ -854,6 +854,70 @@ static bool IsNeighborRoadTile(TileIndex tile, const DiagDirection dir, uint dis
 	return false;
 }
 
+/**
+ * Check whether growing on a half-tile coast tile ends up blocking a water connection
+ *
+ * @param tile The target tile
+ * @return true if building here blocks a water connection
+ */
+static bool GrowingBlocksWaterConnection(TileIndex tile)
```

Comment:
Is this function really the simplest for the test needed?

Reply:
yes